docs/source/notes/cuda.rst

```diff
@@ -264,7 +264,29 @@ used.  For example, the following code is incorrect::
 When the "current stream" is the default stream, PyTorch automatically performs
 necessary synchronization when data is moved around, as explained above.
 However, when using non-default streams, it is the user's responsibility to
-ensure proper synchronization.
+ensure proper synchronization.  The fixed version of this example is::
+
+    cuda = torch.device('cuda')
+    s = torch.cuda.Stream()  # Create a new stream.
+    A = torch.empty((100, 100), device=cuda).normal_(0.0, 1.0)
+    s.wait_stream(torch.cuda.default_stream(cuda))  # NEW!
+    with torch.cuda.stream(s):
+        B = torch.sum(A)
```

Comment:
This is missing a `wait_stream` or `record_stream` call. The "deallocation" of A on the default stream needs to happen *after* the `torch.sum()` reads from A. Or, in more concrete terms, without additional synchronization, the caching allocator may re-use the memory for `A` once it goes out of scope *before* the `torch.sum()` call completes.

(You explain the need for this synchronization in the `record_stream` comment)

```
    cuda = torch.device('cuda')
    s = torch.cuda.Stream()  # Create a new stream.
    A = torch.empty((100, 100), device=cuda).normal_(0.0, 1.0)
    s.wait_stream(torch.cuda.default_stream(cuda))  # NEW!
    with torch.cuda.stream(s):
        B = torch.sum(A)
    torch.cuda.default_stream(cuda).wait_stream(s)
```

or

```
    cuda = torch.device('cuda')
    s = torch.cuda.Stream()  # Create a new stream.
    A = torch.empty((100, 100), device=cuda).normal_(0.0, 1.0)
    s.wait_stream(torch.cuda.default_stream(cuda))  # NEW!
    with torch.cuda.stream(s):
        B = torch.sum(A)
A.record_stream(s)  # location of this call does not really matter
```

Reply:
oh so embarrassing, thanks for catching